### PR TITLE
[WIP] Connector Infor + Account moves export synchronization

### LINKED ISSUE #1: connector_infor
Dependency: connector
License: AGPL-3

# Module description

Using Infor ION (http://infor.com), this module is meant to provide the facility to integrate Odoo with Infor. You should provide your specific integration scenario in a separate module inheriting this one.

There are 2 options to exchange data:

1. SQL: an intermediate database supported by your version of Infor ION
2. File: a file server supporting FTP, SFTP or any protocol supported by storage_backend

## Database
The structure of the database is the one suggested by Infor with inbox and outbox tables. You can find the script to create the structure here: 
[IOBOX script create.zip](https://github.com/OCA/connector-infor/files/1866491/IOBOX.script.create.zip)

## File
The BOD file is transferred to the file server where it is read by Infor Ion and processed.

# Installation

## Database
* Setup a MySQL or MSSQL server that can be accessed by Odoo and Infor ION
* Create a database using the script in [IOBOX script create.zip](https://github.com/OCA/connector-infor/files/1866491/IOBOX.script.create.zip)
* Create 2 users for Odoo and Infor ION and make sure they can connect to the database remotely

## File
* Setup a file server that can be accessed by Odoo and Infor ION using the selected protocol of the file storage

# Configuration
* Go to Settings > Connectors > Infor Backends
* Create a new backend by entering all the info of the intermediate database and the message headers

# Views
backend database
backend headers

### PR DESCRIPTION
**Pull request by guewen**
Replace #3 and #4 
Closes #1 and #2

### CODE REVIEW

File: connector_infor/README.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Images
+------
+
+* Odoo Community Association: `Icon <https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.svg>`_.
+
+Contributors
+------------
+
+* Bhavesh Odedra <bodedra@opensourceintegrators.com>
```
Comment (guewen) [inline connector_infor/README.rst:54]: me and @TDu to add

File: connector_infor_account_move/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+        """Format monetary value."""
+        return '{0:.5f}'.format(n)
+
+    @staticmethod
+    def _default_text(moves):
+        """Default description for some field."""
+        dates = set(moves.mapped('date'))
+        if len(dates) == 1:
+            date = next(iter(dates))
+            return 'STOCK-{}'.format(''.join(date.split('-')))
```
Comment (max3903) [inline connector_infor_account_move/models/account_move.py]: @TDu Let's use the journal short code to prefix the description:
- Stock Journal --> STJ-20180718

**Review by max3903 — CHANGES_REQUESTED** [review]

File: connector_infor_account_move/messages/move.xml.tmpl [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          <JournalStatus>
+            <Code>Final</Code>
+          </JournalStatus>
+        </BaseJournalEntryHeader>
+      </JournalEntryHeader>
+      {% for line in JOURNAL_LINES %}
+      <JournalEntryLine sequence="{{ loop.index }}">
+        <Amount currencyID="{{ line.currency_id.name or COMPANY_CURRENCY|default('', true) }}">{{ "%.5f"|format(amount_currency or line.credit or line.debit|default(0, true)) }}</Amount>
+        <FunctionalAmount currencyID="{{ COMPANY_CURRENCY|default('', true) }}">{{ "%.5f"|format(line.credit or line.debit|default(0, true)) }}</FunctionalAmount>
+        <ReportingCurrencyAmount currencyID="{{ REPORTING_CURRENCY|default('', true) }}">{{ REPORTING_AMOUNT}}</ReportingCurrencyAmount>
```
Comment (max3903) [inline connector_infor_account_move/messages/move.xml.tmpl]: @TDu With the following example:
screenshot from 2018-07-18 16-13-55

I get:
screenshot from 2018-07-18 16-14-17

I am expecting:
```
<Amount currencyID="EUR">100.00000</Amount>
<FunctionalAmount currencyID="USD">83.33000</FunctionalAmount>
```

**Review by max3903 — CHANGES_REQUESTED** [review]

File: connector_infor_account_move/models/account_move.py (hunk 2 above)
Comment (TDu) [reply connector_infor_account_move/models/account_move.py]: Changed

File: connector_infor_account_move/models/account_move.py [hunk 4]
```diff
@@ -120,7 +120,7 @@ def _default_text(moves):
         dates = set(moves.mapped('date'))
         if len(dates) == 1:
             date = next(iter(dates))
-            return 'STOCK-{}'.format(''.join(date.split('-')))
+            return 'STJ-{}'.format(''.join(date.split('-')))
```
Comment (max3903) [inline connector_infor_account_move/models/account_move.py]: @TDu The value should be dynamic and use the short code of the related journal.

If the purchase journal was synced daily, we would expect "PUR-20180719".

File: connector_infor_account_move/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+        """Format monetary value."""
+        return '{0:.5f}'.format(n)
+
+    @staticmethod
+    def _default_text(moves):
+        """Default description for some field."""
+        dates = set(moves.mapped('date'))
+        if len(dates) == 1:
+            date = next(iter(dates))
+            return 'STJ-{}'.format(''.join(date.split('-')))
```
Comment (max3903) [inline connector_infor_account_move/models/account_move.py]: @TDu It should be dynamic to support other journals, not hard-coded.

**Review by max3903 — APPROVED** [review]

File: connector_infor_account_move/models/account_move.py (hunk 5 above)
Comment (TDu) [reply connector_infor_account_move/models/account_move.py]: Oops sorry I misunderstood, fixing

**Pull request merged**